### PR TITLE
ci: configure retries on tests

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

it adds support for retry test with pyTests. (https://github.com/pytest-dev/pytest-rerunfailures)

## Why is it important?

some test fails occasionally but there is not really an error, because of that retries the failed test 3 times with a little sleep of 15 seconds will stabilize the test results on the integration test. 

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -61,6 +62,7 @@ pipeline {
     string(name: 'awsRegion', defaultValue: 'eu-central-1', description: 'Default AWS region to use for testing.')
     booleanParam(name: 'debug', defaultValue: false, description: 'Allow debug logging for Jenkins steps')
     booleanParam(name: 'dry_run', defaultValue: false, description: 'Skip build steps, it is for testing pipeline flow')
+    string(name: 'PYTEST_ADDOPTS', defaultValue: '--reruns 3 --reruns-delay 15', description: 'Aditional options to pass to pyTest. Use PYTEST_ADDOPTS="-k pattern" to only run tests matching the specified pattern.')
```
Comment (cachedout) [inline Jenkinsfile:65]:
```suggestion
    string(name: 'PYTEST_ADDOPTS', defaultValue: '--reruns 3 --reruns-delay 15', description: 'Additional options to pass to pytest. Use PYTEST_ADDOPTS="-k pattern" to only run tests matching the specified pattern.')
```

**Review by cachedout — APPROVED** [review]

**Comment by jsoriano** [comment]
+1 to add retries, but only if we have also something to track flaky tests. If we only add retries we may end-up with more and worse flaky tests, as CI will be less sensible to flakiness. Worst case we may end-up with tests that only sometimes pass and 3 retries won't be enough.

Would this plugin also allow to generate some report on tests that need more retries so we can still act on them?

Maybe another option is to use the `flaky` mark provided by the same plugin, so we can mark a test as flaky as we do now, but without completely skipping it.

**Comment by kuisathaverat** [comment]
>+1 to add retries, but only if we have also something to track flaky tests. If we only add retries we may end-up with more and worse flaky tests, as CI will be less sensible to flakiness. Worst case we may end-up with tests that only sometimes pass and 3 retries won't be enough.

Well, I agree, but 3 runs are not much and this will detect really painful tests we should fix on the first instance, after fix 'em all, we would move the retry to 2, and so on. In that way, we will remove all flaky test and we add a layer of resilience to infrastructure issues.

>Would this plugin also allow to generate some report on tests that need more retries so we can still act on them?

no, it does not support to report the number of runs right now, see https://github.com/pytest-dev/pytest-rerunfailures/issues/97

>Maybe another option is to use the flaky mark provided by the same plugin, so we can mark a test as flaky as we do now, but without completely skipping it.

It is possible to rerun only flaky
https://github.com/pytest-dev/pytest-rerunfailures#re-run-individual-failures

File: Jenkinsfile [hunk 2]
```diff
@@ -61,6 +62,7 @@ pipeline {
     string(name: 'awsRegion', defaultValue: 'eu-central-1', description: 'Default AWS region to use for testing.')
     booleanParam(name: 'debug', defaultValue: false, description: 'Allow debug logging for Jenkins steps')
     booleanParam(name: 'dry_run', defaultValue: false, description: 'Skip build steps, it is for testing pipeline flow')
+    string(name: 'PYTEST_ADDOPTS', defaultValue: '--reruns 3 --reruns-delay 15', description: 'Additional options to pass to pytest. Use PYTEST_ADDOPTS="-k pattern" to only run tests matching the specified pattern.')
```
Comment (kuisathaverat) [inline Jenkinsfile:65]:
```suggestion
    string(name: 'PYTEST_ADDOPTS', defaultValue: '', description: 'Additional options to pass to pytest. Use PYTEST_ADDOPTS="-k pattern" to only run tests matching the specified pattern. For retries you can use `--reruns 3 --reruns-delay 15`')
```

**Comment by kuisathaverat** [comment]
finally, we introduced the plugin and the environment variable empty, so you have to enable retries manually. The plugin allows us to mark tests as flaky and allow retry them.

**Pull request merged**